cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -730,6 +730,33 @@ TEST_P(TestParquetFileFormatScan, PredicatePushdownRowGroupFragmentsUsingDuratio
   CountRowGroupsInFragment(fragment, {0}, expr);
 }
 
+TEST_P(TestParquetFileFormatScan,
+       PredicatePushdownRowGroupFragmentsUsingTimestampColumn) {
+  // GH-37799: Parquet arrow will change TimeUnit::SECOND to TimeUnit::MILLI
+  // because parquet LogicalType don't support SECOND.
+  for (auto time_unit : {TimeUnit::MILLI, TimeUnit::SECOND}) {
+    auto table = TableFromJSON(schema({field("t", time32(time_unit))}),
+                               {
+                                   R"([{"t": 1}])",
+                                   R"([{"t": 2}, {"t": 3}])",
+                               });
+    TableBatchReader table_reader(*table);
+    SCOPED_TRACE(
+        "TestParquetFileFormatScan."
+        "PredicatePushdownRowGroupFragmentsUsingTimestampColumn");
```

Comment:
The name of the test case is already logged by default, scoped traces are usually for providing context from inside the scope of the loop. Did you mean
```suggestion
    ARROW_SCOPED_TRACE("time_unit=", time_unit);
```